test/unit/lib/auth-client.js

```diff
@@ -251,6 +251,117 @@ test('getAuthUrl - test mode false', function(t) {
   t.is(actual, expected);
 });
 
+test('getAuthUrl - single select true', function(t) {
+  const client = new AuthClient({
+    clientId: CLIENT_ID,
+    clientSecret: CLIENT_SECRET,
+    redirectUri: 'https://insurance.co/callback',
+    scope: ['read_odometer', 'read_vehicle_info'],
+  });
+
+  const actual = client.getAuthUrl({
+    scope: 'this should be ignored',
+    state: 'fakestate',
+    forcePrompt: true,
+    singleSelect: true,
+  });
+
+  let expected = 'https://connect.smartcar.com/oauth/authorize?';
+  expected += `response_type=code&client_id=${CLIENT_ID}`;
+  expected += '&redirect_uri=https%3A%2F%2Finsurance.co%2Fcallback';
+  expected += '&approval_prompt=force';
+  expected += '&scope=read_odometer%20read_vehicle_info';
+  expected += '&state=fakestate';
+  expected += '&single_select=true';
+  expected += '&mode=live';
+
+  t.is(actual, expected);
+});
+
+test('getAuthUrl - single select vin', function(t) {
```

Comment:
Maybe add one more tests that includes wrong keys passed in the object?

Reply:
got it!